Sources/TuistGenerator/Mappers/GenerateModuleMapProjectMapper.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+import XcodeProj
+
+/// A project mapper that generates derived modulemap for targets which:
+/// - don't define a modulemap
+/// - contain a single header with a name different from the target name.
+public final class GenerateModuleMapProjectMapper: ProjectMapping {
+    private let derivedDirectoryName: String
+    private let moduleMapsDirectoryName: String
+
+    public init(
+        derivedDirectoryName: String = Constants.DerivedDirectory.name,
+        moduleMapsDirectoryName: String = Constants.DerivedDirectory.moduleMaps
+    ) {
+        self.derivedDirectoryName = derivedDirectoryName
+        self.moduleMapsDirectoryName = moduleMapsDirectoryName
+    }
+
+    // MARK: - ProjectMapping
+
+    public func map(project: Project) throws -> (Project, [SideEffectDescriptor]) {
+        var results = (targets: [Target](), sideEffects: [SideEffectDescriptor]())
+        results = try project.targets.reduce(into: results) { results, target in
```

Comment:
can't this be just `let results: [(targets: [Target], sideEffects: [SidEffectDescriptor])] = try project.target.reduce(into: (targets: [], sideEffects: []))` instead? `results` then does not have to be mutable in the function's scope

Reply:
yep, it was just a copy-paste from the `GenerateInfoPlistProjectMapper`, fixed there as well 🙌